### ISSUE TITLE
CompilePackage: option to compile dependencies

### ISSUE THREAD

**Issue by mtorpey** [gap-packages/PackageManager#58]
As discussed in Issue #49, it's hard to know when to compile dependencies on an update.  Our current approach is to automatically run the build script on packages that cannot be loaded.  Then we have `CompilePackage` as a manual workaround for any packages that are missed.

@james-d-mitchell suggests that it might be nice to be able to compile a package and all its dependencies, since this is something that comes up a lot (again, see Issue #49).  This seems like a reasonable idea, either by an optional argument to `CompilePackage` or by introducing a new function with a name like `CompilePackageAndDependencies`.

Related: how about a `CompileAllPackages` function that runs the script on everything in the user package dir?  Seems like this would be useful for things like updated GAP versions.

**Comment by james-d-mitchell** [comment]
I think you can close this now, you resolved it right?

**Comment by mtorpey** [comment]
Not quite.  We can run `CompilePackage`, but it doesn't compile dependencies automatically, which might be a nice addition later on.
